### PR TITLE
Remove 'static lifetime on key-type trait

### PR DESCRIPTION
**Pull request by richardpringle**
You should be able to use a borrowed key for lookups

### CODE REVIEW

File: firewood/src/v2/api.rs [hunk 1]
```diff
@@ -8,9 +8,9 @@ use async_trait::async_trait;
 /// A `KeyType` is something that can be xcast to a u8 reference,
 /// and can be sent and shared across threads. References with
```
Comment (richardpringle) [inline firewood/src/v2/api.rs:9]: Why do keys need to be shared across threads, futures?

File: firewood/src/v2/api.rs [hunk 2]
```diff
@@ -8,9 +8,9 @@ use async_trait::async_trait;
...
 /// lifetimes are not allowed (hence 'static)
-pub trait KeyType: AsRef<[u8]> + Send + Sync + Debug + 'static {}
+pub trait KeyType: AsRef<[u8]> + Send + Sync + Debug {}
```
Comment (rkuris) [inline firewood/src/v2/api.rs:11]: This is going to break the ownership of keys in a number of places. Maybe instead rename this to OwnedKey and add:
```
pub trait Key: AsRef<[u8]> + Send + Sync + Debug {}
```

Sprinkle `Key` in where it's clear we should be borrowing.

**Review by rkuris — CHANGES_REQUESTED** [review]

File: firewood/src/v2/api.rs (hunk 2 above)
Comment (richardpringle) [reply firewood/src/v2/api.rs:11]: It's not breaking anything. `KeyType + 'static` is a subset of `KeyType` and should be added where necessary, not the other way around.

Comment (rkuris) [reply firewood/src/v2/api.rs:11]: OK that makes sense. Do we need to add any `'static`s anywhere?

**Review by rkuris — APPROVED** [review]

**Pull request merged**